src/IIIFPresentation/API/Helpers/BatchHelper.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Models.Database.Collections;
+using Models.Database.General;
+using Repository;
+
+namespace API.Helpers;
+
+public static class BatchHelper
+{
+    public static async Task SaveBatchesInDlcs(this List<DLCS.Models.Batch> batches, 
+        Manifest manifest, PresentationContext dbContext, CancellationToken cancellationToken = default)
+    {
+        var dbBatches = batches.Select(b => new Models.Database.General.Batch
+        {
+            Id = Convert.ToInt32(b.ResourceId!.Split('/').Last()),
+            CustomerId = manifest.CustomerId,
+            Submitted = b.Submitted.ToUniversalTime(),
+            Status = BatchStatus.Ingesting,
+            ManifestId = manifest.Id
+        });
+        
+        await dbContext.Batches.AddRangeAsync(dbBatches, cancellationToken);
+        await dbContext.SaveChangesAsync(cancellationToken);
```

Comment:
By calling `.SaveChanges()` here you will be saving everything that has been added to the ef context so far. Looking at `ManifestService` this should go via the `SaveAndPopulateEntity` method as that calls the `TrySave()` method which has specific error handling for duplicate slugs - if that error was thrown here you'd lose the nice error message + problemJson response type.